### PR TITLE
Automatically format with clang-tools

### PR DESCRIPTION
**Pull request by grahamc**
Sort of a shitpost, but most times I read through Nix I wish it was consistently formatted.

### CODE REVIEW

**Comment by lovesegfault** [comment]
I was dying for something like this when I tried to write some code for Nix a few days ago. Hope it happens!

File: src/libcmd/command.cc [hunk 1]
```diff
@@ -211,60 +207,58 @@ Strings editorFor(const Path & file, uint32_t line)
...
 {
-    addFlag({
-        .longName = "profile",
-        .description = "The profile to update.",
-        .labels = {"path"},
-        .handler = {&profile},
-        .completer = completePath
-    });
+    addFlag(
+        {.longName = "profile",
```
Comment (edolstra) [inline src/libcmd/command.cc:221]: The change to the formatting of designated initializers causes a huge diff, but I don't see a way to keep the old style...

File: src/nix/log.cc [hunk 2]
```diff
@@ -17,11 +17,14 @@ struct CmdLog : InstallableCommand
     std::string doc() override
     {
         return
-          #include "log.md"
-          ;
+#include "log.md"
```
Comment (edolstra) [inline src/nix/log.cc:20]: This is ugly, it should follow the indentation.

**Review by thufschmitt — CHANGES_REQUESTED** [review]
The diff is obviously scary, but overall I think it's a great thing!

One thing that would be needed imho to make this really useful is a way to enforce this in the long run as it will quickly diverge from that style again. Maybe a CI job checking the format of everything?

Also, if the expectation is to enforce this, we should make it clear and easy for contributors by updating the hacking docs accordingly and having a nice `make` target for it (I know there's already a script for it, but I don't think it's as discoverable and nice to use)

File: src/libcmd/command.cc (hunk 1 above)
Comment (fricklerhandwerk) [reply src/libcmd/command.cc:221]: We could use [`.git-blame-ignore-revs`](https://github.com/community/community/discussions/5033#discussioncomment-2318478) as [we do in `nixpkgs`](https://github.com/NixOS/nixpkgs/blob/master/.git-blame-ignore-revs).

**Comment by edolstra** [comment]
TODO: add a CI check to verify formatting.

**Pull request closed without merging**